test/controllers/organizations/pets_controller_test.rb

```diff
@@ -57,4 +58,33 @@ class Organizations::PetsControllerTest < ActionDispatch::IntegrationTest
     assert_equal Mime[:html], response.media_type
     assert_response :redirect
   end
+
+  test "POST default pet tasks are created when pet is created" do
+    post pets_path, params: post_params
```

Comment:
```suggestion
    assert_difference "Pet.count", 1 do
       post pets_path, params: post_params
    end
```
This is going to better ensure that we are certainly creating a Pet and then you can called `created_pet = Pet.last` and do your assertions.

Reply:
Done